lib/langchain/llm/openai.rb

```diff
@@ -44,7 +44,8 @@ def initialize(api_key:, llm_options: {}, default_options: {})
         top_logprobs: {},
         n: {default: @defaults[:n]},
         temperature: {default: @defaults[:temperature]},
-        user: {}
+        user: {},
+        response_format: {default: @defaults[:response_format]}
```

Comment:
FYI: if you add a default, it will use it (even if it is `nil`).  If you made this just an empty hash it should work to solve #766 
```suggestion
        response_format: {}
```

Reply:
The reason for this is that the unified parameters will drop any argument without a value _unless_ it has a default value (which it will use).

Reply:
@codenamev The problem is that initializing the client with the default_options does not then set those parameters on every consecutive call:
```ruby
irb(main):002> llm = Langchain::LLM::OpenAI.new(api_key: ENV["OPENAI_API_KEY"], default_options: { response_format: { type: "json_object" } })
=>
#<Langchain::LLM::OpenAI:0x0000000125ebc690
```

The `response_format: { type: "json_object" }` parameter is not automatically set in the following call:
```ruby
irb(main):003> llm.chat messages:[{role:"user", content:"hey json!"}]
=>
#<Langchain::LLM::OpenAIResponse:0x0000000125b7c3e0
 @model=nil,
 @raw_response=
  {"id"=>"chatcmpl-ABAha7eKlGmTw9QRSoPQPCAvoqe5E",
   ...
   "choices"=>[{"index"=>0, "message"=>{"role"=>"assistant", "content"=>"Hello! How can I assist you today?", "refusal"=>nil}, "logprobs"=>nil, "finish_reason"=>"stop"}],
```

Reply:
I see. The initializer should be able to mutate the library defaults.  I'll audit the LLM subclasses today.